src/ixdat/techniques/deconvolution.py

```diff
@@ -0,0 +1,254 @@
+                    data.
+                EC_data (list): List of numpy arrays containing the EC (time,
+                    current, potential).
+        """
+
+        if MS_data is not None and parameters:
+            raise Exception(
+                'Kernel can only be initialized with data OR parameters, not both'
+                )
+        if EC_data is not None and MS_data is not None:
+            print('Generating kernel from measured data')
+            self.type = 'measured'
+        elif parameters:
+            print('Generating kernel from parameters')
+            self.type = 'functional'
+        else:
+            print('Generating blank kernel')
+            self.type = None
+
+        self.params = parameters
+        self.MS_data = MS_data
+        self.EC_data = EC_data #x_curr, y_curr, x_pot, y_pot
+
+    @property
+    def sig_area(self):
+        """Integrates a measured impulse response and returns the area."""
+        delta_sig = self.MS_data[1] - self.MS_data[1][-1]
+        sig_area = np.trapz(delta_sig, self.MS_data[0])
+
+        return sig_area
+
+
+    @property
+    def charge(self):
+        """Integrates the measured current over the time of the impulse and returns
+        the charge passed.
+        """
+        y_curr = self.EC_data[1]
+
+        mask = np.isclose(
+            y_curr,
+            y_curr[0],
+            rtol=1e-1
+            )
+
+        Q = np.trapz(y_curr[mask], self.EC_data[0][mask])
+
+        return Q
+
+    def plot(self,
+        time=None,
+        ax = None,
+        norm=True,
+        **kwargs
+        ):
+        """Returns a plot of the kernel/impulse response."""
+        if ax is None:
+            fig1 = plt.figure()
+            ax = fig1.add_subplot(111)
+
+        if self.type is 'functional':
+            ax.plot(
+                time, self.gen_kernel(time=time, norm=norm),
+                **kwargs,
+                )
+
+        elif self.type is 'measured':
+            ax.plot(
+            self.MS_data[0], self.gen_kernel(norm=norm),
+            **kwargs,
+            )
+
+        else:
+            raise Exception('Nothing to plot with blank kernel')
+
+
+        return ax
+
+    def gen_kernel(self,
+        dt=0.1,
+        len=100,
+        norm=True,
+        matrix=False
+        ):
+        """Generates a kernel/impulse response based on the defined mass
+        transport parameters contained in dict.
+
+        Args:
+            dt (int): Timestep for which the kernel/impulse response is calculated.
+            len(int): Timelength in seconds for which the kernel/impulse response is
+                calculated.
+            norm (bool): If true the kernel/impulse response is normalized to its
+                area.
+            matrix (bool): If true the circulant matrix constructed from the kernel/
+                impulse reponse is returned.
+        """
+        if self.type is 'functional':
+
+            time = np.arange(0,len,dt)
```

Comment:
I don't like the name `time` for a numpy array. It indicates to me something grander, like the `time` module. 
`t` is in my opinion perfect for this, with an underscore and a discription whenever that could be ambiguous. Here it could be `t_kernel`.